### PR TITLE
geticons: init at 1.2.2

### PR DESCRIPTION
**Pull request by Madouura**
## Description of changes
For ricing purposes

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ocfox — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5484,6 +5484,8 @@ with pkgs;
 
   gdu = callPackage ../tools/system/gdu { };
 
+  geticons = callPackage ../by-name/ge/geticons/package.nix { };
+
```
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:5488]:
```suggestion
```
Packages found in the named-based structure do not need to be explicitly added to the `top-level/all-packages.nix` file unless they require overriding the default value of an implicit attribute.

**Review by ocfox — CHANGES_REQUESTED** [review]

**Review by ocfox — APPROVED** [review]

File: pkgs/by-name/ge/geticons/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    hash = "sha256-HEnUfOLeRTi2dRRqjDPVwVVHo/GN9wE28x5qv3qOpCY=";
+  };
+
+  cargoHash = "sha256-Znwni7uMnG9cpZbztUMY1j73K+XrDLv5zyNEZDoxWg4=";
+  passthru.updateScript = gitUpdater { };
+
+  meta = with lib; {
+    description = "CLI utility to get icons for apps on your system or other generic icons by name";
+    homepage = "https://git.sr.ht/~zethra/geticons";
+    license = with licenses; [ gpl3Only ];
```
Comment (delroth) [inline pkgs/by-name/ge/geticons/package.nix:24]: Cargo.toml seems to indicate GPL-3.0-or-later?

Comment (Madouura) [reply pkgs/by-name/ge/geticons/package.nix:24]: I probably just looked at the LICENSE file and figured it was GPL-3.0-Only.

**Review by l0b0 — APPROVED** [review]
LGTM, thanks. I don't know geticons though, and I don't have merge rights.

**Pull request merged**